### PR TITLE
AutomatticLogo: Introduce Types

### PR DESCRIPTION
**Pull request by ockham**
#### Changes proposed in this Pull Request

AutomatticLogo: Introduce Types

#### Testing instructions

TBD

### CODE REVIEW

File: client/components/automattic-logo/index.tsx [hunk 1]
```diff
@@ -1,15 +1,19 @@
...
 import './style.scss';
 
-const AutomatticLogo = ( { className, size } ) => (
+interface Props {
+	className?: string;
+	size?: number;
+}
+
+const AutomatticLogo: FunctionComponent< Props > = ( { className = '', size = 96 } ) => (
 	<svg className={ `automattic-logo ${ className }` } height={ size } viewBox="0 0 792 96">
```
Comment (sirreal) [inline client/components/automattic-logo/index.tsx]: I don't like that with the default we wind up with class `automattic-logo ` (note trailing space). How about something like this:

```suggestion
	<svg className={ 'automattic-logo' + ( className ? className : '' ) } height={ size } viewBox="0 0 792 96">
```

File: client/components/automattic-logo/index.tsx [hunk 2]
```diff
@@ -1,15 +1,19 @@
...
  */
 import './style.scss';
 
-const AutomatticLogo = ( { className, size } ) => (
+interface Props {
+	className?: string;
+	size?: number;
+}
+
+const AutomatticLogo: FunctionComponent< Props > = ( { className = '', size = 96 } ) => (
```
Comment (sirreal) [inline client/components/automattic-logo/index.tsx]: The default `""` seems silly.

```suggestion
const AutomatticLogo: FunctionComponent< Props > = ( { className, size = 96 } ) => (
```

**Review by sirreal — APPROVED** [review]
Thanks!

This is used in the footer for Crowdsignal SSO - I got there by logging in from https://crowdsignal.com/ and swapping in the calypso.live url.

Testing well.

File: client/components/automattic-logo/index.tsx (hunk 1 above)
Comment (ockham) [reply client/components/automattic-logo/index.tsx]: But now `'automattic-logo'` and the `className` are no longer separated by a space at all, are they?

Comment (ockham) [reply client/components/automattic-logo/index.tsx]: Good old

```suggestion
	<svg className={ compact( [ 'automattic-logo', className ] ).join( ' ' ) } height={ size } viewBox="0 0 792 96">
```

maybe?

Comment (ockham) [reply client/components/automattic-logo/index.tsx]: Or use the `classNames` package? :grimacing:

Comment (ockham) [reply client/components/automattic-logo/index.tsx]: 5dd5366

**Review by sirreal — APPROVED** [review]
Looks good. E2e had a canary failure, let's figure out if that's cause for concern before landing (I don't see how it could be related, but 🤷‍♂ )

**Pull request merged**